agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4397,15 +4397,62 @@ bool slave_thread::handle_autoconfiguration_wsc(Socket *sd, ieee1905_1::CmduMess
         }
     }
 
-    if (cmdu_rx.getNextTlvType() != uint8_t(ieee1905_1::eTlvType::TLV_VENDOR_SPECIFIC)) {
-        LOG(ERROR) << "Not vendor specific TLV (not Intel?)";
-        return false;
-    }
+    if (cmdu_rx.getNextTlvType() == uint8_t(ieee1905_1::eTlvType::TLV_VENDOR_SPECIFIC)) {
+        LOG(INFO) << "Intel controller join response";
+        if (!parse_intel_join_response(sd, cmdu_rx)) {
+            LOG(ERROR) << "Parse join response failed";
+            return false;
+        }
+    } else {
+        LOG(INFO) << "Non-Intel controller join response";
+        if (slave_state != STATE_WAIT_FOR_JOINED_RESPONSE) {
+            LOG(ERROR) << "slave_state != STATE_WAIT_FOR_JOINED_RESPONSE";
+            return false;
+        }
 
-    LOG(INFO) << "Intel controller join response";
-    if (!parse_intel_join_response(sd, cmdu_rx)) {
-        LOG(ERROR) << "Parse join response failed";
-        return false;
+        // request the current vap list from ap_manager
+        auto request = message_com::create_vs_message<
+            beerocks_message::cACTION_APMANAGER_HOSTAP_VAPS_LIST_UPDATE_REQUEST>(cmdu_tx);
+        if (request == nullptr) {
+            LOG(ERROR)
+                << "Failed building cACTION_APMANAGER_HOSTAP_VAPS_LIST_UPDATE_REQUEST message!";
+            return false;
+        }
+        message_com::send_cmdu(ap_manager_socket, cmdu_tx);
+
+        // send all pending_client_association notifications
+        for (auto notify : pending_client_association_cmdu) {
+            auto notification = message_com::create_vs_message<
+                beerocks_message::cACTION_CONTROL_CLIENT_ASSOCIATED_NOTIFICATION>(cmdu_tx);
+            if (notification == nullptr) {
+                LOG(ERROR) << "Failed building message!";
+                return false;
+            }
+            notification->params() = notify.second;
+            send_cmdu_to_controller(cmdu_tx);
+        }
+        pending_client_association_cmdu.clear();
+
+        // No version checking for non-Intel controller
```

Comment:
Agree - so why keep the TODO?

Reply:
I don't understand your question. The TODO is about passing the master/slave versions to the platform manager, where it will eventually be used by bml_get_master_slave_versions. The "No version checking" comment is about not checking the version (i.e. the version is always considered correct).